### PR TITLE
cleanup(spanner): tidy up spanner backup/restore samples

### PR DESCRIPTION
**Pull request by devbww**
This is mostly to ease the upcoming merge of backup/restore
samples for PITR and CMEK, which include some general fixes
like eliminating unused using declarations, adding missing
newlines, and correcting tag names.  Make those changes here
first so that the future merges are all about the features.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5874)

### CODE REVIEW

File: google/cloud/spanner/samples/samples.cc [hunk 1]
```diff
@@ -1779,7 +1770,7 @@ void QueryNewColumn(google::cloud::spanner::Client client) {
     std::cout << "AlbumId: " << std::get<1>(*row) << "\t";
     auto marketing_budget = std::get<2>(*row);
     if (marketing_budget) {
-      std::cout << "MarketingBudget: " << marketing_budget.value() << "\n";
+      std::cout << "MarketingBudget: " << *marketing_budget << "\n";
```
Comment (coryan) [inline google/cloud/spanner/samples/samples.cc:1773]: nit: unrelated?

**Review by coryan — APPROVED** [review]

File: google/cloud/spanner/samples/samples.cc (hunk 1 above)
Comment (devbww) [reply google/cloud/spanner/samples/samples.cc:1773]: Yes.  And no ... I should have mentioned in the description an attempt to make the samples more `StatusOr`-idiomatic.  The underlying issue is that I probably should not have attempted any cleanup in a feature branch, but it is difficult to resist.
